neo4j-migrations-core/src/test/java/ac/simons/neo4j/migrations/core/Neo4jImageNameSubstitutor.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2020-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ac.simons.neo4j.migrations.core;
+
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.ImageNameSubstitutor;
+
+/**
+ * Changes the source repository to aarch64 compatible images for neo4j < 4.4
+ */
+public class Neo4jImageNameSubstitutor extends ImageNameSubstitutor {
+
+	@Override
+	public DockerImageName apply(DockerImageName dockerImageName) {
+		// don't change ryuk and neo4j 4.4+ coordinates
+		if (!dockerImageName.getRepository().startsWith("neo4j") || (dockerImageName.getRepository().startsWith("neo4j") && dockerImageName.getVersionPart().contains("4.4"))) {
+			return dockerImageName;
+		}
+		String property = System.getProperty("os.arch", "");
```

Comment:
Be careful, you need to check Docker's architecture, not host's.

`DockerClientFactory.instance().getInfo().getArchitecture()` would be a much better thing to check 👍

Reply:
Wohooo..great. Was looking for something like this but then picked the option with the env var :D 
Thanks a lot, Sergei.